enumerables.rb

```diff
@@ -0,0 +1,137 @@
+module Enumerable
+  def my_each
+    i = 0
+    arr = self
+    if block_given?
+      while i < arr.count
+        yield(arr[i])
+        i += 1
+      end
+    else
+      arr
+    end
+  end
+
+  def my_each_with_index
+    i = 0
+    arr = self
+    if block_given?
+      while i < arr.count
+        yield(arr[i], i)
+        i += 1
+      end
+    else
+      arr
+    end
+  end
+
+  def my_select
+    i = 0
+    to_change = []
+    arr = self
+    if block_given?
+      while i < arr.count
+        to_change << arr[i] if yield(arr[i])
+        i += 1
+      end
+      to_change
+    else
+      arr
+    end
+  end
+
+  def my_any?(arg = nil)
+    arr = self
+    i = 0
+    to_change = false
+    while i < arr.count
+      if !block_given?
+        to_change = true if include?(true) && arg.nil?
+        to_change = true if arr[i] == arg && arg.respond_to?(:to_i)
+        to_change = true if arr[i].match(arg) && arg.is_a?(Regexp)
+        to_change = true if arr[i].instance_of?(arg) && arg.respond_to?(:class)
+      else
+        to_change = false unless yield(arr[i])
+      end
+      i += 1
+    end
+    to_change
+  end
+
+  def my_all?
+    i = 0
+    to_change = true
+    arr = self
+    while i < arr.count
+      to_change = false unless yield(arr[i])
+      i += 1
+    end
+    to_change
+  end
+
+  def my_none?
+    i = 0
+    to_change = true
+    arr = self
+    while i < arr.count
+      to_change = false if yield(arr[i])
+      i += 1
+    end
+    to_change
+  end
+
+  def my_map(the_proc)
+    i = 0
+    to_change = []
+    arr = self
+    arg = the_proc.nil?
+    while i < arr.count
+      to_change << (the_proc.call arr[i]) unless arg == true
+      to_change << yield(arr[i]) if arg == true
+      i += 1
+    end
+    to_change
+  end
+
+  def my_count
+    arr = self
+    i = 0
+    total = 0
+    while i < arr.count
+      arr.my_each { total += 1 }
+      i += 1
+    end
+    total
+  end
+
+  def my_inject(start_value = 0, operator = nil)
```

Comment:
- [ ] my_inject should also accept a block. Make to add that condition.

- [ ] What happens if there's no start_value and only an operator is used as the sole operator? Make sure it works the same as [1,2,3,4,5].inject(:+) does.